capirca/lib/arista_tp.py

```diff
@@ -0,0 +1,1036 @@
+        # does not match.
+        if self.term.platform:
+            if self._PLATFORM not in self.term.platform:
+                return ""
+        if self.term.platform_exclude:
+            if self._PLATFORM in self.term.platform_exclude:
+                return ""
+
+        config = Config()
+
+        # a LoL which will be appended to the config at the end of this method
+        # elements will be of the form [indentation, string, verbatim] by
+        # default verbatim = False
+        term_block = []
+
+        # don't render icmpv6 protocol terms under inet, or icmp under inet6
+        if (self.term_type == "inet6" and "icmp" in self.term.protocol) or (
+            self.term_type == "inet" and "icmpv6" in self.term.protocol
+        ):
+            logging.debug(
+                self.NO_AF_LOG_PROTO.substitute(
+                    term=self.term.name,
+                    proto=", ".join(self.term.protocol),
+                    af=self.term_type,
+                )
+            )
+            return ""
+
+        # term verbatim output - this will skip over normal term creation
+        # code.  warning generated from policy.py if appropriate.
+        if self.term.verbatim:
+            for line in self.term.verbatim:
+                if line[0] == self._PLATFORM:
+                    # pass MATCH_INDENT, but this should be ignored in the
+                    # rendering
+                    term_block.append([MATCH_INDENT, str(line[1]), True])
+            # we return immediately, there's no action to be formed
+            for i, s, v in term_block:
+                config.Append(i, s, verbatim=v)
+
+            return str(config)
+
+        # option processing
+        flags = []
+        misc_options = []
+        if self.term.option:
+            flags, misc_options = self._processTermOptions(self.term,
+                                                           self.term.option)
+
+        # helper for per-address-family keywords.
+        family_keywords = self._TERM_TYPE.get(self.term_type)
+
+        # if the term name is default-* we will render this into the appropriate
+        # default term name to be used in this filter.
+        default_term = re.match(r"^default\-.*", self.term.name,
+                                re.IGNORECASE)
+        if default_term:
+            self.term.name = family_keywords["addr_fam"] + "-default-all"
+
+        term_block.append(
+            [TERM_INDENT,
+             "match %s %s" % (self.term.name, family_keywords["addr_fam"]),
+             False
+             ]
+        )
+
+        term_af = self.AF_MAP.get(self.term_type)
+        # comment - could use a little formatting love
+        if self.term.owner and not self.noverbose:
+            self.term.comment.append("owner: %s" % self.term.owner)
+        if self.term.comment and not self.noverbose:
+            for comment in self.term.comment:
+                for line in comment.split("\n"):
```

Comment:
This could do with some sort of comment length truncation/splitting into multi-line comments. 
Large comments would end up in a single indented line.

Is there a comment width limit? If so, you can split on that basis, otherwise, do what is native to the platform for multi-line comments. If there's nothing specified by the platform, I would recommend splitting after 80 characters.

Reply:
https://github.com/google/capirca/blob/master/doc/generator_patterns.md#perform-truncating-of-names-for-term-name-or-comment-based-on-max-length is the corresponding requirement.